cli.c: do not use iobuf pool

mykaul opened the pull request:
There's really no need to allocate an iobuf pool for the CLI.
It's a waste of memory and CPU to do all this allocation for CLI command
invocation.

Fixes: #2963
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
cli.c: do not use iobuf pool

There's really no need to allocate an iobuf pool for the CLI.
It's a waste of memory and CPU to do all this allocation for CLI command
invocation.

Fixes: #2963
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mykaul commented:
/run regression

mykaul commented:
/run regression

mohit84 commented on cli/src/cli.c:
```diff
@@ -224,6 +216,32 @@ logging_init(glusterfs_ctx_t *ctx, struct cli_state *state)
     return 0;
 }
 
+static struct iobuf *
```
I think instead of writing a new function we can directly call iobuf_get_from_small and make a function non-static.

mykaul replied:
Will do!

Commit:
iobuf.c: make iobuf_get_from_small() non static and use in cli

Re-use the function, by making it non-static.

(Remove the unneeded use of 'ret' in the function while at it).

Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mykaul commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

0 test(s) generated core 


15 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_changelog_changelog-snapshot.t
./tests/000-flaky/basic_distribute_rebal-all-nodes-migrate.t
./tests/000-flaky/basic_ec_ec-quorum-count-partial-failure.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t
./tests/000-flaky/bugs_distribute_bug-1117851.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_glusterd_bug-857330/normal.t
./tests/000-flaky/bugs_glusterd_bug-857330/xml.t
./tests/000-flaky/bugs_glusterd_quorum-value-check.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

14 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_changelog_changelog-snapshot.t
./tests/000-flaky/basic_distribute_rebal-all-nodes-migrate.t
./tests/000-flaky/basic_ec_ec-quorum-count-partial-failure.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t
./tests/000-flaky/bugs_distribute_bug-1117851.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_glusterd_bug-857330/normal.t
./tests/000-flaky/bugs_glusterd_bug-857330/xml.t
./tests/000-flaky/bugs_glusterd_quorum-value-check.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
./tests/000-flaky/glusterd-restart-shd-mux.t
https://build.gluster.org/job/gh_centos7-regression/1850/

mohit84 commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

0 test(s) generated core 


15 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_changelog_changelog-snapshot.t
./tests/000-flaky/basic_distribute_rebal-all-nodes-migrate.t
./tests/000-flaky/basic_ec_ec-quorum-count-partial-failure.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t
./tests/000-flaky/bugs_distribute_bug-1117851.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_glusterd_bug-857330/normal.t
./tests/000-flaky/bugs_glusterd_bug-857330/xml.t
./tests/000-flaky/bugs_glusterd_quorum-value-check.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/00-georep-verify-non-root-setup.t

14 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_changelog_changelog-snapshot.t
./tests/000-flaky/basic_distribute_rebal-all-nodes-migrate.t
./tests/000-flaky/basic_ec_ec-quorum-count-partial-failure.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t
./tests/000-flaky/bugs_distribute_bug-1117851.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_glusterd_bug-857330/normal.t
./tests/000-flaky/bugs_glusterd_bug-857330/xml.t
./tests/000-flaky/bugs_glusterd_quorum-value-check.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
./tests/000-flaky/glusterd-restart-shd-mux.t
https://build.gluster.org/job/gh_centos7-regression/1851/

mykaul commented:
/run regression

Commit:
Merge branch 'devel' into cli_iobuf

mykaul commented:
/run regression
